openflow_input/bsn_port_desc_module_eeprom

```diff
@@ -0,0 +1,228 @@
+//
+// You may not use this file except in compliance with the EPL or LOXI Exception. You may obtain
+// a copy of the EPL at:
+//
+// http://www.eclipse.org/legal/epl-v10.html
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// EPL for the specific language governing permissions and limitations
+// under the EPL.
+#version 5
+#version 6
+
+enum ofp_bsn_unit_type(wire_type=uint8_t, bitmask=False) {
+    OFP_BSN_UNIT_NONE = 0x0,
+    OFP_BSN_UNIT_AMP = 0x2,
+    OFP_BSN_UNIT_VOL = 0x3,
+    OFP_BSN_UNIT_KM = 0x4,
+    OFP_BSN_UNIT_M = 0x5,
+    OFP_BSN_UNIT_S = 0x6,
+    OFP_BSN_UNIT_C = 0x7,
+    OFP_BSN_UNIT_F = 0x8
+};
+
+struct of_bsn_unit {
+    uint32_t value;
+    /* We could probably use an enum to convey the unit */
+    enum ofp_bsn_unit_type unit;
+};
+
+enum ofp_bsn_module_eeprom_recv_signal_type(wire_type=uint8_t, bitmask=False) {
+    OFP_BSN_MODULE_EEPROM_RECV_SIGNAL_OMA = 0x0,
+    OFP_BSN_MODULE_EEPROM_RECV_SIGNAL_AVG = 0x1
+};
+
+enum ofp_bsn_module_eeprom_diag_type(wire_type=uint8_t, bitmask=False) {
+    OFP_BSN_MODULE_EEPROM_DIAG_CURR = 0x0,
+    OFP_BSN_MODULE_EEPROM_DIAG_VOLT = 0x1,
+    OFP_BSN_MODULE_EEPROM_DIAG_TEMP = 0x2,
+    OFP_BSN_MODULE_EEPROM_DIAG_RECP = 0x3,
+    OFP_BSN_MODULE_EEPROM_DIAG_OUTP = 0x4
+};
+
+enum ofp_bsn_module_eeprom_identifier(wire_type=uint8_t, bitmask=False) {
+    OFP_BSN_MODULE_EEPROM_IDENTIFIER_NONE = 0x0,
+    OFP_BSN_MODULE_EEPROM_IDENTIFIER_GBIC = 0x1,
+    OFP_BSN_MODULE_EEPROM_IDENTIFIER_ONBOARD = 0x2,
+    OFP_BSN_MODULE_EEPROM_IDENTIFIER_SFP = 0x3
+};
+
+enum ofp_bsn_module_eeprom_extended_identifier(wire_type=uint8_t, bitmask=False) {
+    OFP_BSN_MODULE_EEPROM_EXT_IDENTIFIER_NON_GBIC = 0x00,
+    OFP_BSN_MODULE_EEPROM_EXT_IDENTIFIER_GBIC_SFP = 0x04,
+    OFP_BSN_MODULE_EEPROM_EXT_IDENTIFIER_GBIC_MOD_DEF = 0x07
+};
+
+enum ofp_bsn_module_eeprom_connector(wire_type=uint8_t, bitmask=False) {
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_NONE = 0x00,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_SC = 0x01,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_FC_1_COPPER = 0x02,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_FC_2_COPPER = 0x03,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_BNC_TNC = 0x04,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_FC_COAX = 0x05,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_FIBERE_JACK = 0x06,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_LC = 0x07,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_MT_RJ = 0x08,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_MU = 0x09,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_SG = 0x0a,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_OPTICAL_PIGTAIL = 0x0b,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_MPO_PARALLEL_OPTIC = 0x0c,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_HSSDC = 0x20,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_COPPER_PIGTAIL = 0x21,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_RJ45 = 0x22
```

Comment:
are these values specified elsewhere, and you are just enumerating only the ones you expect to see ?